File: src/com/nerdscorner/android/plugin/github/ui/tablemodels/GHReleaseTableModel.java
package com.nerdscorner.android.plugin.github.ui.tablemodels;

import java.io.Serializable;
import java.text.SimpleDateFormat;
import java.util.List;

import com.nerdscorner.android.plugin.github.domain.gh.GHReleaseWrapper;
import com.nerdscorner.android.plugin.utils.Strings;

public class GHReleaseTableModel extends BaseModel<GHReleaseWrapper> implements Serializable {
    private static final int COLUMN_TAG = 0;
    private static final int COLUMN_DATE = 1;

    public GHReleaseTableModel(List<GHReleaseWrapper> releases, String[] colNames) {
        super(releases, colNames);
    }

    @Override
    public Object getValueAt(int rowIndex, int columnIndex) {
        if (rowIndex < 0 || rowIndex >= items.size()) {
            return null;
        }
        GHReleaseWrapper release = items.get(rowIndex);
        switch (columnIndex) {
            case COLUMN_TAG:
                return release.getGhRelease().getName();
            case COLUMN_DATE:
                return new SimpleDateFormat(Strings.DATE_FORMAT).format(release.getGhRelease().getPublished_at());
        }
        return null;
    }
}

File: src/com/nerdscorner/android/plugin/github/ui/tablemodels/GHReleaseTableModel.kt
package com.nerdscorner.android.plugin.github.ui.tablemodels

import com.nerdscorner.android.plugin.github.domain.gh.GHReleaseWrapper
import java.io.Serializable

class GHReleaseTableModel(releases: MutableList<GHReleaseWrapper>, colNames: Array<String>)
    : BaseModel<GHReleaseWrapper>(releases, colNames), Serializable {

    override fun getValueAt(rowIndex: Int, columnIndex: Int): Any? {
        if (rowIndex < 0 || rowIndex >= items.size) {
            return null
        }
        val release = items[rowIndex]
        return when (columnIndex) {
            COLUMN_TAG -> release.ghRelease.name
            COLUMN_DATE -> release.publishedAt
            else -> null
        }
    }

    companion object {
        const val COLUMN_TAG = 0
        const val COLUMN_DATE = 1
    }
}
